modules/excoBidAdapter.js

```diff
@@ -1,37 +1,299 @@
-  return `https://${subDomain}.exco-pb.com`;
+    bidResponse.cpm = bid.cpm;
+
+    const { bidRequest } = context;
+
+    bidResponse.width = bid.w || deepAccess(bidRequest, 'mediaTypes.video.w') || deepAccess(bidRequest, 'params.video.playerWidth');
+    bidResponse.height = bid.h || deepAccess(bidRequest, 'mediaTypes.video.h') || deepAccess(bidRequest, 'params.video.playerHeight');
+
+    if (deepAccess(bid, 'ext.bidder.rp.advid')) {
+      deepSetValue(bidResponse, 'meta.advertiserId', bid.ext.bidder.rp.advid);
+    }
+
+    return bidResponse;
+  }
+
+  log(severity, message) {
+    const msg = `${BIDDER_CODE.toUpperCase()}: ${message}`;
+
+    if (severity === 'warn') {
+      logWarn(msg);
+    }
+    if (severity === 'error') {
+      logError(msg);
+    }
+    if (severity === 'info') {
+      logInfo(msg);
+    }
+  }
 }
 
-const buildRequests = createBuildRequestsFn(createDomain, null, storage, BIDDER_CODE, BIDDER_VERSION, false);
+const helpers = new AdapterHelpers();
+
+/**
+ * @description https://github.com/prebid/Prebid.js/blob/master/libraries/ortbConverter/README.md
+ */
+export const converter = ortbConverter({
+  request(buildRequest, imps, bidderRequest, context) {
+    const data = buildRequest(imps, bidderRequest, context);
+
+    if (data.cur && !data.cur.includes('USD')) {
+      helpers.log('warn', 'Warning - EX.CO adapter is supporting USD only. processing with USD');
+    }
+
+    data.cur = [CURRENCY];
+    data.test = config.getConfig('debug') ? 1 : 0;
+
+    helpers.addOrtbFirstPartyData(data, context.bidRequests || []);
+
+    return data;
+  },
+  imp(buildImp, bidRequest, context) {
+    const imp = buildImp(bidRequest, context);
+
+    imp.secure = window.location.protocol === 'http:' ? 0 : 1;
+
+    if (imp.video) {
+      helpers.adoptVideoImp(imp, bidRequest)
+    }
 
-const interpretResponse = createInterpretResponseFn(BIDDER_CODE, false);
+    if (imp.banner) {
+      helpers.adoptBannerImp(imp, bidRequest);
+    }
 
-const getUserSyncs = createUserSyncGetter({
-  iframeSyncUrl: 'https://cs.exco-pb.com/api/sync/iframe', imageSyncUrl: 'https://cs.exco-pb.com/api/sync/image'
+    return imp;
+  },
+  bidResponse(buildBidResponse, bid, context) {
+    const bidResponse = buildBidResponse(bid, context);
+    return helpers.adoptBidResponse(bidResponse, bid, context);
+  },
+  context: {
+    ttl: 3000,
+  },
+  processors: pbsExtensions
 });
 
 export const spec = {
   code: BIDDER_CODE,
-  version: BIDDER_VERSION,
-  supportedMediaTypes: [BANNER, VIDEO],
-  gvlid: GVLID,
-  isBidRequestValid,
-  buildRequests,
-  interpretResponse,
-  getUserSyncs
+  supportedMediaTypes: [VIDEO, BANNER],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {import('../src/auction.js').BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function (bid) {
```

Comment:
this appears to suddenly expect new parameters. This looks breking, marking as prebid 10